src/audio/render_thread.rs

```diff
@@ -117,7 +121,12 @@ impl AudioRenderThread {
                 }
                 // and push into the audio sink the result of processing a
                 // render quantum.
-                let _ = self.sink.push_data(self.process());
+                if let Ok(duration) = self.sink.push_data(self.process()) {
```

Comment:
we shouldn't need to ask the sink for the duration; we know how long it is. The sink calculates the duration from information on the buffer that _we_ provided.